actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -105,182 +107,225 @@ def generate_content_security_policy_nonce
         end
     end
 
-    MAPPINGS = {
-      self:           "'self'",
-      unsafe_eval:    "'unsafe-eval'",
-      unsafe_inline:  "'unsafe-inline'",
-      none:           "'none'",
-      http:           "http:",
-      https:          "https:",
-      data:           "data:",
-      mediastream:    "mediastream:",
-      blob:           "blob:",
-      filesystem:     "filesystem:",
-      report_sample:  "'report-sample'",
-      strict_dynamic: "'strict-dynamic'",
-      ws:             "ws:",
-      wss:            "wss:"
-    }.freeze
-
-    DIRECTIVES = {
-      base_uri:        "base-uri",
-      child_src:       "child-src",
-      connect_src:     "connect-src",
-      default_src:     "default-src",
-      font_src:        "font-src",
-      form_action:     "form-action",
-      frame_ancestors: "frame-ancestors",
-      frame_src:       "frame-src",
-      img_src:         "img-src",
-      manifest_src:    "manifest-src",
-      media_src:       "media-src",
-      object_src:      "object-src",
-      prefetch_src:    "prefetch-src",
-      script_src:      "script-src",
-      script_src_attr: "script-src-attr",
-      script_src_elem: "script-src-elem",
-      style_src:       "style-src",
-      style_src_attr:  "style-src-attr",
-      style_src_elem:  "style-src-elem",
-      worker_src:      "worker-src"
-    }.freeze
-
-    DEFAULT_NONCE_DIRECTIVES = %w[script-src style-src].freeze
-
-    private_constant :MAPPINGS, :DIRECTIVES, :DEFAULT_NONCE_DIRECTIVES
-
-    attr_reader :directives
+    attr_reader :formats
 
     def initialize
-      @directives = {}
+      @formats = Hash.new { |h,k| h[k] = Format.new }
       yield self if block_given?
     end
 
     def initialize_copy(other)
-      @directives = other.directives.deep_dup
+      @formats = other.formats.deep_dup
     end
 
-    DIRECTIVES.each do |name, directive|
-      define_method(name) do |*sources|
-        if sources.first
-          @directives[directive] = apply_mappings(sources)
-        else
-          @directives.delete(directive)
+    (Mime::SET.map(&:to_sym) + [:any]).each do |type|
```

Comment:
Do we need to generate methods for all the formats? I think we should only generate for the most common ones (html, js, json, css, any) and let the other formats be specified in another method like `formats.any(:xml)`